core/slim_sim.cpp

```diff
@@ -6269,14 +6269,6 @@ void SLiMSim::AddIndividualsToTable(Individual * const *p_individual, size_t p_n
 					   == (p_tables->individuals.metadata_offset[tsk_individual + 1]
 						   - p_tables->individuals.metadata_offset[tsk_individual])));
 			
-			// BCH 4/29/2019: This assert is, we think, not technically necessary – the code
-			// would work even if it were violated.  But it's a nice invariant to guarantee,
-			// and right now it is always true.
-			assert((p_tables->nodes.individual[ind->genome1_->tsk_node_id_]
-					 == tsk_individual)
-				   && (p_tables->nodes.individual[ind->genome2_->tsk_node_id_]
-					   == tsk_individual));
-			
```

Comment:
This bit of code is duplicated above, and there doesn't seem to be the possibility that the node tables are changed between there and here?

Reply:
If you're referring to the line above where it does `p_tables->nodes.individual[ind->genome1_->tsk_node_id_] = tsk_individual;`, that is in a different branch of the if/else statement, isn't it?

Reply:
Sorry, I meant ["below", here](https://github.com/MesserLab/SLiM/pull/159/files/d2b23c05f7e0e2a1642154a1f22d0560e1dbb113#diff-0ae7a04e87e1ce83b282da28825cc9492ad328d2d4a6d0343fb24f61137de991R6294).

Reply:
Figured I'd keep the consistency check that's at the end.

Reply:
Ah, I see.  Yes, the first check seems redundant, I suppose.  I can't remember why we did that.